File: esports-backend/src/main/java/com/esports/dto/UpdateRoleRequest.java
package com.esports.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
public class UpdateRoleRequest {
    @NotNull(message = "User ID is required")
    private Long userId;
    
    @NotBlank(message = "Role is required")
    private String role;
}

File: Upkart1/DTO/UpdateRoleRequest.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Upkart1.DTO
{
    public class UpdateRoleRequest
    {
        [Required(ErrorMessage = "User ID is required")]
        [JsonPropertyName("userId")]
        public long UserId { get; set; }
        
        [Required(ErrorMessage = "Role is required")]
        [JsonPropertyName("role")]
        public string Role { get; set; }
    }
}